CRM/Dedupe/Merger.php

```diff
@@ -1914,7 +1914,7 @@ public static function getMergeCacheKeyString($rule_group_id, $group_id, $criter
     $cacheKeyString = "merge {$contactType}";
     $cacheKeyString .= $rule_group_id ? "_{$rule_group_id}" : '_0';
     $cacheKeyString .= $group_id ? "_{$group_id}" : '_0';
-    $cacheKeyString .= !empty($criteria) ? serialize($criteria) : '_0';
+    $cacheKeyString .= !empty($criteria) ? md5(serialize($criteria)) : '_0';
```

Comment:
Probably a silly question, but from the description it looks like this key is stored in the database. When upgrading to the next version will the cache always be cleared? It would probably be harmless to have the old version of the cache key still in the database anyway, but just thought I'd check.

Reply:
I think upgrades do clear these out, or you can schedule a job to intermittently clear cache tables.

This is the sort of cache that just gets regenerated if empty. I can't see any risks on there being extraneous or missing entries